PostHog/PostHogSDK.swift

```diff
@@ -468,7 +468,10 @@ let maxRetryDelay = 30.0
 
         let isIdentified = storageManager.isIdentified()
 
-        if distinctId != oldDistinctId, !isIdentified {
+        let hasDistinctIdChanges = distinctId != oldDistinctId && !isIdentified
+        let hasUserPropertyChanges = !(userProperties?.isEmpty ?? true) || !(userPropertiesSetOnce?.isEmpty ?? true)
```

Comment:
the benefit of keeping this check below is that its only executed if the first if is false, right now we do the check even if not needed, its a lil perf improvement, but your call